top_view.py

```diff
@@ -31,13 +31,13 @@ def wait90TopviewThread(gov_robot, prefix1,prefix90):
     time.sleep(0.10)
   try:
     topViewWrite()
-    topViewSnap(prefix90,os.getcwd()+"/pinAlign",1)
+    topViewSnap(prefix90,getBlConfig("visitDirectory")+"/pinAlign",1)
     snapshot1Name = prefix1+"_001.jpg"
     snapshot2Name = prefix90+"_001.jpg"
-    if (not filecmp.cmp(os.getcwd()+"/pinAlign/"+snapshot1Name,os.getcwd()+"/pinAlign/"+snapshot2Name)): #this would mean something is wrong if true because the pictures are identical
+    if (not filecmp.cmp(getBlConfig("visitDirectory")+"/pinAlign/"+snapshot1Name,getBlConfig("visitDirectory")+"/pinAlign/"+snapshot2Name)): #this would mean something is wrong if true because the pictures are identical
       comm_s = os.environ["LSDCHOME"] + "/runPinAlign.py " + snapshot1Name + " " + snapshot2Name
       logger.info(comm_s)
-      lines = os.popen(comm_s).readlines()
+      lines = os.popen(comm_s, cwd=getBlConfig("visitDirectory")).readlines()
```

Comment:
Other places are using `os.popen` which weren't updated. Check with:

```bash
$ git grep -n os.popen
beamline_lib.py:314:  for outputline in os.popen(comm_s).readlines():
daq_macros.py:867:    resultString = "<data>\n"+os.popen(comm_s).read()+"</data>\n"
daq_macros.py:2590:    for outputline in os.popen(comm_s).readlines():
daq_utils.py:295:  masterFilename = os.popen(comm_s).read()[0:-1]
runPinAlign.py:14:lines = os.popen(comm_s).readlines()
runSpotFinder4syncW.py:55:  resultString = "<data>\n"+os.popen(comm_s).read()+"</data>\n"
top_view.py:40:      lines = os.popen(comm_s, cwd=getBlConfig("visitDirectory")).readlines()
```

Reply:
Most of them look innocuous, but the daq_utils:295 one does act as the base for filenames that could be used as output.

Reply:
I can apply these changes to this pull request if you want them made here.

Reply:
@mrakitin would you be happy if all of the `os.environ["PWD"]` were changed to `getBlConfig("visitDirectory")` and the `os.popen` calls were kept as-is for now? the justification for keeping them is that our tests have already shown that for the code that is actually callable during our current operations, all files have been written to the correct place.

Reply:
@mskinner5278 can you do the task mentioned in this thread, just the `os.environ["PWD"]` one? once that looks good, I'd like to merge and then you can rebase future branches on top of that

Reply:
I will do that now.

Reply:
@mskinner5278 is going to prepare a new pull request with the fixes because he was not able to push the commits onto my existing branch. resolving conversation